BDArmory/Settings/CompSettings.cs

```diff
@@ -0,0 +1,76 @@
+using System.Linq;
+using System.Reflection;
+using UnityEngine;
+
+using System.IO;
+using System.Collections.Generic;
+using System;
+using BDArmory.UI;
+
+namespace BDArmory.Settings
+{
+	public class CompSettings
+	{
+		// Settings overrides for AI/WM settings for competition rules compliance
+
+		static readonly string CompSettingsPath = Path.GetFullPath(Path.Combine(KSPUtil.ApplicationRootPath, "GameData/BDArmory/PluginData/Comp_settings.cfg"));
+		static public bool CompOverridesEnabled = false;
+		static readonly Dictionary<string, object> ComPOverrides = new()
+		{
+				// FIXME there's probably a few more things that could get set here for AI/WM override if needed in specific rounds.
+				//AI Min/max Alt?
+				{"PS_EXTEND_TIMEOUT", -1},
```

Comment:
I was thinking that these should be completely separate from the main `BDArmorySettings...` settings so that when using them you'd access them via `CompSettings.CompOverrides["override_name"]` (or `CompSettings.OverrideName` or `CustomOverrides.OverrideName`) and they wouldn't overwrite any settings in the main settings file. The `Load` function would then just be something similar to `BDAPersistentSettingsField.Load`.

Being separate from the rest of the settings would mean we could also drop the `PS_` prefix to make it less specific to project showdown.

--------------------------

If we wanted to make this even more generic so that code changes wouldn't be needed for future overrides (at least that don't require extra checks), then the `typeof(Class).GetField(...)` stuff could be moved to the `ApplyFJRT` (or `ApplyCustomOverrides`) function (which might make sense to move to this file and just call it from the `SpawnUtilsInstance` instance) and the values applied via reflection.

The custom_overrides.cfg (or whatever it ends up being called) would then be something like
```
CustomOverrides
{
  BDModulePilotAI
  {
    idleSpeed = 42;
    ...
  }
  MissileFire
  {
    CockpitFOV = 120; 270 
    ...
  }
}
```
though that's probably a bit complex for the current needs.

Reply:
Done a half-way house implementation for now; they're now completely offloaded from settings.cfg and entirely comp_settings.cfg-based, but not to the point of GetField reflection. Not sure if going all the way for that would be needed, though; looking through the various AI and Wm settings, there's only a few more that'd maybe make sense to have rules compliance overrides for.

Reply:
Yeah, the reflection-based approach is probably overkill.